scripts/querystring-rider.js

```diff
@@ -0,0 +1,54 @@
+
+
+// Update Query String properties on each page
+
+(function(){
+    function gid(id) {
+      return document.getElementById(id);
+    }
+    function qsa(s) {
+      return document.querySelectorAll(s);
+    }
+    function forEach(array, callback, scope) {
+      for (var i = 0; i < array.length; i++) {
+        callback.call(scope || array[i], i, array[i]);
+      }
+    }
+
+    var qs = tinyQuery.getAll();
+
+    if (qs.souls2thepolls) {
+        var intro = gid('intro');
+        if (intro) {
+          intro.classList.add('s2tp');
+        }
+
+        var logo = gid('logo');
+        logo.setAttribute('src', '/images/logo-souls2thepolls.png');
+        logo.setAttribute('width', 661);
+        logo.setAttribute('height', 300);
+
+        forEach(qsa('a[href^="/"]'), function() {
+          this.href = tinyQuery.set('souls2thepolls', false, this.href);
+        });
+
+        forEach(qsa('.twitter-mention-button'), function() {
+          var url = this.getAttribute('data-url');
+          this.setAttribute('data-url', tinyQuery.set('souls2thepolls', false, url));
+        });
+    }
+
+    if (qs.uuid) {
+        forEach(qsa('.uuid'), function() {
+            this.textContent = qs.uuid;
+        });
+
+        forEach(qsa('.self-service-url'), function() {
+            // this.href = tinyQuery.set(qs, this.href);
+            this.href += "?UUID_rider=" + qs.uuid;
```

Comment:
I see that this is fixing an issue on the site where the server redirect is assuming no existing query strings, and is adding a `?` when it adds the `uuid` value, resulting in URLs like `http://carpoolvote.com/thanks-rider/?type=rider?uuid=468c4ade-a196-4368-9136-7e017748dc6d`.

However instead of duplicating the entire query-string file twice, a DRYer solution would be to add a data-attribute to each `.self-service-url` link, specifying the `type`. e.g.:
HTML:
```
<input type="hidden" name="_redirect" class="redirect" value="http://carpoolvote.com/thanks-rider/" data-type="rider" />
```
JS:
```
forEach(qsa('.self-service-url'), function() {
    qs.type = this.getAttribute('data-type');
    qs['uuid_' + qs.type] = qs.uuid;
    this.href = tinyQuery.set(qs, this.href);
});
```